src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs

```diff
@@ -1359,60 +1365,173 @@ private BoundExpression GetDefaultParameterSpecialNoConversion(SyntaxNode syntax
                 if (parameter is not null)
                 {
                     visitedParameters[parameter.Ordinal] = true;
+
+                    if (expanded && parameter.Ordinal == paramsIndex)
+                    {
+                        Debug.Assert(arrayArgsBuilder is not null);
+                        Debug.Assert(arrayArgsBuilder.Count == 0);
+
+                        firstParamArrayArgument = i;
+                        arrayArgsBuilder.Add(argumentsBuilder[i]);
+
+                        for (int remainingArgument = i + 1; remainingArgument < argumentsBuilder.Count; ++remainingArgument)
+                        {
+                            if (GetCorrespondingParameter(remainingArgument, parameters, argsToParamsOpt, expanded: true)?.Ordinal != paramsIndex)
+                            {
+                                break;
+                            }
+
+                            arrayArgsBuilder.Add(argumentsBuilder[remainingArgument]);
+                            i++;
```

Comment:
> Looks like we could use `i` in this loop directly instead of `remainingArgument` since we are incrementing it here anyway.

I do not think so. The `i` is incremented conditionally.